src/main/java/org/mule/service/oauth/internal/builder/AbstractOAuthDancerBuilder.java

```diff
@@ -120,6 +121,19 @@ public OAuthDancerBuilder tokenUrl(String tokenUrl, TlsContextFactory tlsContext
     return this;
   }
 
+  @Override
+  public OAuthDancerBuilder<D> tokenUrl(String tokenUrl, TlsContextFactory tlsContextFactory, ProxyConfig proxyConfig) {
+    this.tokenUrl = tokenUrl;
+    this.httpClientFactory = () -> {
+      final Builder clientConfigBuilder =
+          new HttpClientConfiguration.Builder().setName(format("oauthToken.requester[%s]", tokenUrl));
+      clientConfigBuilder.setTlsContextFactory(tlsContextFactory);
```

Comment:
what about the case where you have a proxy but not a tlsContext? does this work?

Reply:
Sure. The test cases that I've added cover this scenario